x-pack/plugins/security_solution/public/management/pages/host_isolation_exceptions/view/host_isolation_exceptions_list.tsx

```diff
@@ -56,6 +57,25 @@ export const HostIsolationExceptionsList = () => {
   const location = useHostIsolationExceptionsSelector(getCurrentLocation);
   const navigateCallback = useHostIsolationExceptionsNavigateCallback();
 
+  const [memoizedRouteState, setMemoizedRouteState] = useState<ListPageRouteState | undefined>();
+  useEffect(() => {
+    if (routeState && routeState.onBackButtonNavigateTo) {
+      setMemoizedRouteState(routeState);
+    }
+  }, [routeState]);
```

Comment:
Looks like this could be made into a generic hook that we can then use any page. We can talk more about it offline. I'm thinking the hook could accept a "url route prefix" that it would continue to check for it to be "truthy" and the moment its not, the memoized route state would be reset back to `undefined`.